pyccel/codegen/printing/ccode.py

```diff
@@ -175,7 +186,11 @@ def _print_PyccelAdd(self, expr):
         return ' + '.join(self._print(a) for a in expr.args)
 
     def _print_PyccelMinus(self, expr):
-        return ' - '.join(self._print(a) for a in expr.args)
+        args = [self._print(a) for a in expr.args]
+
+        if len(args) == 1:
+            return '-{}'.format(args[0])
+        return ' - '.join(a for a in args)
```

Comment:
Why not just `return ' - '.join(args)`?

Could you give me an example where `PyccelMinus` has more than one argument?

Reply:
Do you mean where `PyccelMinus` has one argument? The example is in issue #399
The unary operator `-x` is translated to `PyccelMinus(PyccelUnary(x))`

Reply:
I mean a situation where `len(args)>1`, which triggers line 193.

Reply:
Then `x - y` = PyccelMinus(x,y)

Reply:
Sorry I didn't spot the difference between
`return ' - '.join(args)` and
`return ' - '.join(a for a in args)`

Reply:
OK, so the `args` tuple can either contain 1 or 2 elements here. Never 3 I suppose? I just find the code a little strange, as they way it is written makes one think that `args` could have an arbitrary length.

Reply:
No never 3 for a binary operator (boolean operators on the other hand can contain as many elements as they want despite not appearing significantly different in construction)

Reply:
In the official Python documentation an explicit distinction is made between unary and binary math operators. Here `PyccelMinus` covers both, is this the same in Python's `ast`?

Reply:
No. Python's `ast` has a `BinaryOp` class and a `UnaryOp` class. Each of these has an operator eg Add, Sub, UAdd, USub, Not, etc.
Pyccel's treatment of Unary operators is a bit odd:
```
        if isinstance(stmt.op, ast.Not):
            Func = PyccelNot
        elif isinstance(stmt.op, ast.UAdd):
            Func = PyccelUnary
        elif isinstance(stmt.op, ast.USub):
            Func = PyccelMinus

        return Func(PyccelUnary(target))
```
I'm not sure why the second `PyccelUnary` is necessary other than to ensure that the correct precedence is used. I think it would be better to have a `PyccelUnaryMinus`. I also don't think that Not should be using `PyccelUnary` as it has a much lower precedence. This function certainly wants looking at. But it is almost certainly beyond the scope of this PR

Reply:
Yes, this is beyond the scope of this PR (in fact I am keeping the discussion as "resolved")... but it's probably useful to keep track of our conversation here. Maybe we just need to create a `PyccelUnaryMinus` class, but we could also mimic Python's `ast` and have `PyccelBinaryOp` and `PyccelUnaryOp`.

Reply:
Yes, that is also a possibility. It is currently written this way as it mimics sympy (although sympy doesn't have Minus or Divide). The current method makes specifying the precedence easy though.

Yes we should create an issue